index.js

```diff
@@ -17,7 +19,13 @@ module.exports = function (str, filepath) {
 
 	return pify(mkdirp, Promise)(path.dirname(fullpath))
 		.then(function () {
-			return pify(fs.writeFile, Promise)(fullpath, str);
+			return isStream(str) ?
+				new Promise((resolve, reject) => {
+					str.pipe(fs.createWriteStream(fullpath))
```

Comment:
You also need to attach an error handler to the write stream, as errors (annoyingly) do not propagate in streams.

```js
str.pipe(fs.createWriteStream(fullpath).on('error', reject))
```

Can you also create the promise outside the ternary? Makes it more readable.